src/server.cpp

```diff
@@ -2126,6 +2139,61 @@ void Server::sendAddNode(v3s16 p, MapNode n, u16 ignore_id,
 	}
 }
 
+void Server::sendMetadataChanged(const std::list<v3s16> &meta_updates, float far_d_nodes)
+{
+	float maxd = far_d_nodes * BS;
+	NodeMetadataList *meta_updates_list = new NodeMetadataList();
+
+	std::vector<u16> clients = m_clients.getClientIDs();
+	m_clients.lock();
+
+	for (std::vector<u16>::iterator i = clients.begin();
+			i != clients.end(); ++i) {
+		meta_updates_list->clear();
+		Player *player = m_env->getPlayer(*i);
+		RemoteClient* client = m_clients.lockedGetClientNoEx(*i);
+		if (client == NULL)
+			continue;
+		if (client->net_proto_version >= 28) {
+			for (std::list<v3s16>::const_iterator i2 = meta_updates.begin();
+					i2 != meta_updates.end(); ++i2) {
+				v3s16 pos = *i2;
+				NodeMetadata *meta = m_env->getMap().getNodeMetadata(pos);
+				if (!meta) {
+					continue;
+				}
+				if (player) {
```

Comment:
It needs to be inside the inner loop unfortunately as it depends on pos. The only thing that can be put outside is the `v3f player_pos` calculation, but thats not worth it IMO.